File: Back-Ends/JavaScript/src/Module/Group/Model/group.model.js
const newDate = new Date().toISOString();

export default (sequelize, DataTypes) => {
  const Groups = sequelize.define("groups", {
    userId: {
      type: DataTypes.STRING,
      allowNull: false,
      references: {
        model: "Users",
        key: "id",
      },
      unique: false,
      onUpdate: "CASCADE",
      onDelete: "CASCADE",
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    description: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    category: {
      type: DataTypes.ENUM(
        "Trips",
        "Movies",
        "Functions",
        "Bills",
        "Travel",
        "Other",
        "Dinner",
      ),
      allowNull: false,
    },
    groupBalance: {
      type: DataTypes.STRING,
      defaultValue: 0.0,
      allowNull: false,
    },
    goalBalance: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    currency: {
      type: DataTypes.STRING,
      defaultValue: "USD",
      allowNull: false,
    },
    emails: {
      type: DataTypes.ARRAY(DataTypes.STRING),
      allowNull: false,
      unique: false,
    },
    status: {
      type: DataTypes.ENUM("In Progress", "Completed"),
      allowNull: false,
    },
    createdAt: {
      type: newDate,
      allowNull: false,
    },
    updatedAt: {
      type: newDate,
      allowNull: false,
    },
  });
  return Groups;
};

File: Back-Ends/Golang/api/models/group.model.go
package models

import (
	"time"

	pq "github.com/lib/pq"
)

type Group struct {
	ID           uint32         `gorm:"primary_key;auto_increment" json:"id"`
	UserId       uint32         `sql:"type:int REFERENCES users(id)" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Name         string         `gorm:"size:100;not null;" json:"name"`
	Description  string         `gorm:"size:100;not null;" json:"description"`
	Category     string         `gorm:"size:100;not null;" json:"category"`
	GroupBalance string         `gorm:"size:100;not null;default:0" json:"groupBalance"`
	GoalBalance  string         `gorm:"size:100;not null;default:0" json:"goalBalance"`
	Emails       pq.StringArray `gorm:"type:varchar(255)[]"`
	Status       string         `gorm:"size:100;not null;" json:"status"`
	CreatedAt    time.Time      `gorm:"default:CURRENT_TIMESTAMP" json:"createdAt"`
	UpdatedAt    time.Time      `gorm:"default:CURRENT_TIMESTAMP" json:"updatedAt"`
}
